Azure/azure-cli
# [AppService] `az functionapp create`: Use stacks API netFrameworkVersion value

**Related command**
`az functionapp create`

**Description**
Currently, Windows dotnet apps use the [default netFrameworkVersion from the SDK](https://github.com/Azure/azure-sdk-for-python/blob/9056bd8899e449964bd3ae3b3cff3714ba7318c7/sdk/appservice/azure-mgmt-web/azure/mgmt/web/v2023_01_01/models/_models_py3.py#L20910C1-L20910C1)
The reason why we have to access additionalProperties to fetch netFrameworkVersion is because [netFrameworkVersion is not part of SiteConfigPropertiesDictionary](https://github.com/Azure/azure-sdk-for-python/blob/9056bd8899e449964bd3ae3b3cff3714ba7318c7/sdk/appservice/azure-mgmt-web/azure/mgmt/web/v2023_01_01/models/_models_py3.py#L21252-L21265) in the SDK. 
Resolves #23827

**Testing Guide**
Create Windows dotnet app and make sure that the netFrameworkVersion set is the one from the [stacks API](https://learn.microsoft.com/en-us/rest/api/appservice/provider/get-function-app-stacks?view=rest-appservice-2023-01-01)

**History Notes**


[AppService] `az functionapp create`: Use stacks API netFrameworkVersion value instead of the default value from the Python SDK

---

This checklist is used to make sure that common guidelines for a pull request are followed.
